awx/sso/conf.py

```diff
@@ -1215,6 +1215,44 @@ def _register_ldap(append=None):
     placeholder=SOCIAL_AUTH_TEAM_MAP_PLACEHOLDER,
 )
 
+###############################################################################
+# Generic OIDC AUTHENTICATION SETTINGS
+###############################################################################
+
+register(
+    'SOCIAL_AUTH_OIDC_KEY',
+    field_class=fields.CharField,
+    allow_null=False,
+    default=None,
+    label=_('OIDC Key'),
+    help_text='The OIDC key (Client ID) from your IDP.',
+    category=_('Generic OIDC'),
+    category_slug='oidc',
+)
+
+register(
+    'SOCIAL_AUTH_OIDC_SECRET',
+    field_class=fields.CharField,
+    allow_blank=True,
+    default='',
+    label=_('OIDC Secret'),
+    help_text=_('The OIDC secret (Client Secret) from your IDP.'),
+    category=_('Generic OIDC'),
+    category_slug='oidc',
+    encrypted=True,
+)
+
+register(
+    'SOCIAL_AUTH_OIDC_OIDC_ENDPOINT',
+    field_class=fields.CharField,
+    allow_blank=True,
+    default='',
+    label=_('OIDC Provider URL'),
+    help_text=_('The URL for your OIDC provider, e.g.: http(s)://hostname/.'),
+    category=_('Generic OIDC'),
+    category_slug='oidc',
+)
+
```

Comment:
There is one other setting we may want to think about adding here:
```
register(
    'SOCIAL_AUTH_OIDC_VERIFY_SSL',
    field_class=fields.BooleanField,
    default=True,
    label=_('Verify OIDC Provider Certificate'),
    help_text=_('Verify the OIDV provider ssl certificate.'),
    category=_('Generic OIDC'),
    category_slug='oidc',
)
```
I'm a little torn because we don't do this for other providers but we also occasionally get questions about how to connect to a provider with an untrusted cert. This field may be a quick and easy way for an admin to do some testing. @AlanCoding any thoughts on adding this field?

Reply:
I 100% agree that OIDC should always be SSL and trusted in production. But people also use AWX for testing/development and personal use. We also allow credentials and other items to ignore SSL verification. And the stack trace when SSL is not trusted is pretty massive. I talked with an architect this morning and they are good with us adding this. If we wanted to we could make this just settable via the API and not have a UI field on the form.

Reply:
Let's ignore this for now in this PR and I can add this as an option later as needed.

Reply:
@john-westcott-iv I just added it..... 😆 Want me to revert?

Reply:
If you got it in there, its totally fine; thanks!
We'll get this over to our QE team and hopefully we can merge soon.